### PR TITLE
rust 1.22.1

### PR DESCRIPTION
**Pull request by leonklingele**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

**Comment by leonklingele** [comment]
Any idea why the builds fail? 😩

**Comment by alex** [comment]
not sure if it's related, but you may want to jump to 1.22.1

File: Formula/rust.rb [hunk 1]
```diff
@@ -53,14 +53,13 @@ class Rust < Formula
 
   resource "cargobootstrap" do
     # From https://github.com/rust-lang/rust/blob/#{version}/src/stage0.txt
-    url "https://static.rust-lang.org/dist/2017-08-31/cargo-0.21.0-x86_64-apple-darwin.tar.gz"
-    sha256 "05720fe91343c6ed5c4995f7b06677a08a585209f6f504d192722f3bb585bc9d"
+    url "https://static.rust-lang.org/dist/2017-11-22/cargo-0.23.0-x86_64-apple-darwin.tar.gz"
```
Comment (alex) [inline Formula/rust.rb]: The version you have here doesn't appear to match what's in the file pointed to by the comment

Comment (leonklingele) [reply Formula/rust.rb]: Fixed.

**Comment by leonklingele** [comment]
Updated to latest 1.22.1 release, build still breaks :(

```
--- stderr
thread 'main' panicked at '

Internal error occurred: Command "clang" "-O2" "-ffunction-sections" "-fdata-sections" "-fPIC" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-stdlib=libc++" "--target=x86_64-apple-darwin" "-fno-builtin" "-fvisibility=hidden" "-fomit-frame-pointer" "-ffreestanding" "-DVISIBILITY_HIDDEN" "-o" "/tmp/rust-20171123-17339-16x3yg7/rustc-1.22.1-src/build/x86_64-apple-darwin/stage0-std/x86_64-apple-darwin/release/build/compiler_builtins-fea1b6393f79ff72/out/../../libcompiler_builtins/compiler-rt/lib/builtins/absvdi2.o" "-c" "../../libcompiler_builtins/compiler-rt/lib/builtins/absvdi2.c" with args "clang" did not execute successfully (status code exit code: 1).

', .brew_home/.cargo/registry/src/github.com-1ecc6299db9ec823/cc-1.0.0/src/lib.rs:1645:4
note: Run with `RUST_BACKTRACE=1` for a backtrace.

warning: build failed, waiting for other jobs to finish...
error: build failed
thread 'main' panicked at 'command did not execute successfully: "/tmp/rust-20171123-17339-16x3yg7/rustc-1.22.1-src/build/x86_64-apple-darwin/stage0/bin/cargo" "build" "--target" "x86_64-apple-darwin" "--release" "--features" "panic-unwind jemalloc backtrace" "--manifest-path" "/tmp/rust-20171123-17339-16x3yg7/rustc-1.22.1-src/src/libstd/Cargo.toml" "--message-format" "json"
expected success, got: exit code: 101', src/bootstrap/compile.rs:883:8
note: Run with `RUST_BACKTRACE=1` for a backtrace.
failed to run: /tmp/rust-20171123-17339-16x3yg7/rustc-1.22.1-src/build/bootstrap/debug/bootstrap build
Build completed unsuccessfully in 0:05:52
make: *** [all] Error 1
```

**Comment by ilovezfs** [comment]
I believe this is an upstream bug. Because Homebrew now requires the CLT for <= Sierra, it no longer universally sets SDKROOT or MACOSX_DEPLOYMENT_TARGET, leaving that to formula to do manually when it actually fixes something. This is one of those cases where something breaks without MACOSX_DEPLOYMENT_TARGET being set.

The package `compiler_builtins` doesn't set its own deployment target correctly on its own.
```
error: failed to run custom build command for `compiler_builtins v0.0.0 (file:///private/tmp/rust-20171122-34531-1fyemqj/rustc-1.21.0-src/src/rustc/compiler_builtins_shim)`
process didn't exit successfully: `/private/tmp/rust-20171122-34531-1fyemqj/rustc-1.21.0-src/build/x86_64-apple-darwin/stage0-std/release/build/compiler_builtins-9a4d14296558989f/build-script-build` (exit code: 101)

...

running: "clang" "-O2" "-ffunction-sections" "-fdata-sections" "-fPIC" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-stdlib=libc++" "--target=x86_64-apple-darwin" "-fno-builtin" "-fvisibility=hidden" "-fomit-frame-pointer" "-ffreestanding" "-DVISIBILITY_HIDDEN" "-o" "/private/tmp/rust-20171122-34531-1fyemqj/rustc-1.21.0-src/build/x86_64-apple-darwin/stage0-std/x86_64-apple-darwin/release/build/compiler_builtins-7c6a641db161520a/out/../../libcompiler_builtins/compiler-rt/lib/builtins/absvdi2.o" "-c" "../../libcompiler_builtins/compiler-rt/lib/builtins/absvdi2.c"
cargo:warning=clang: error: invalid deployment target for -stdlib=libc++ (requires OS X 10.7 or later)
exit code: 1
```

A downstream fix is here: https://github.com/Homebrew/homebrew-core/pull/20966

But it technically shouldn't be something we need to do.

**Comment by ilovezfs** [comment]
This needs to be squashed to one commit and the racer resource needs to be bumped too.

**Comment by ilovezfs** [comment]
also please rebase on master.

**Comment by ilovezfs** [comment]
@leonklingele I took care of it.

**Comment by ilovezfs** [comment]
Thanks again, @leonklingele!

**Pull request closed without merging**